File: PRACTICA AUXI/java/INTRODUCCION A LA PROGRAMACION/Persona.java
///"NOMBRE: QUENTA MAMANI NICK RANDY"
/// "CI: 10037571"
///
/// "1. Crea una clase Persona con nombre, edad y ciudad"
/// "a) Agrega un método para mostrar el saludo: “Hola, soy {nombre} de {ciudad}”"
/// "b) Crea tres personas y muestra su saludo"
/// "c) Agrega un método para verificar si es mayor de edad"

public class Persona {
    private String nombre;
    private int edad;
    private String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    public String saludo() {
        return "Hola, soy " + nombre + " de " + ciudad + ".";
    }

    public String verificarMayorEdad() {
        return edad >= 18 ? "es mayor de edad" : "no es mayor de edad";
    }

    public static void main(String[] args) {
        Persona[] personas = {
                new Persona("Mauricio", 30, "Beni"),
                new Persona("Nayra", 16, "El Alto"),
                new Persona("Gafi", 21, "Sucre")
        };

        for (Persona persona : personas) {
            System.out.println(persona.saludo());
            System.out.println(persona.nombre + " " + persona.verificarMayorEdad());
        }
    }
}

File: PRACTICA AUXI/python/INTRODUCCION A LA PROGRAMACION/Persona.py
"NOMBRE: QUENTA MAMANI NICK RANDY"
"CI: 10037571"

"1. Crea una clase Persona con nombre, edad y ciudad"
"a) Agrega un método para mostrar el saludo: “Hola, soy {nombre} de {ciudad}”"
"b) Crea tres personas y muestra su saludo"
"c) Agrega un método para verificar si es mayor de edad"

class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad

    def saludo(self):
        return f"Hola, soy {self.nombre} de {self.ciudad}"

    def es_mayor_de_edad(self):
        return "es mayor de edad" if self.edad >= 18 else "no es mayor de edad"

pers1 = Persona("Mauricio", 32, "Beni")
pers2 = Persona("Nayra", 17, "El Alto")
pers3 = Persona("Gafi", 21, "Sucre")

print(pers1.saludo())
print(pers2.saludo())
print(pers3.saludo())

print(f"{pers1.nombre} {pers1.es_mayor_de_edad()}")
print(f"{pers2.nombre} {pers2.es_mayor_de_edad()}")
print(f"{pers3.nombre} {pers3.es_mayor_de_edad()}")
